CONTRIBUTING.md

```diff
@@ -1,112 +1,124 @@
+# Prerequisites
 
-It can be compiled using the including `Makefile` or executed just with the `go run` command. `rhc`is part of RHEL systems, it can be used instead of using `yggdrasil`.
-This requires `pub` and `sub` modules. To get these modules:
+Being a yggdrasil worker, `rhc-worker-playbook` requires a similar development
+environment to
+[`yggdrasil`](https://github.com/RedHatInsights/yggdrasil/blob/main/CONTRIBUTING.md).
+It is recommended that you first ensure you have a working yggdrasil development
+set up, although it is possible to develop `rhc-worker-playbook` using a
+packaged version of yggdrasil.
 
-```console
-$ go get git.sr.ht/~spc/mqttcli
-$ go install git.sr.ht/~spc/mqttcli/cmd/...
-```
+`rhc-worker-playbook` executes playbooks that assume they are running as a
+privileged user. As a result, the included systemd service runs as `root`.
 
-*MQTT broker*: Needed to publish messages. [Mosquitto](https://mosquitto.org/) is easy to set up and an excellent broker that can be run offline and locally. The online [solution](https://test.mosquitto.org/) can be used but is not recommended.
+## `yggd`
 
+Make sure `yggd` is running and listening on the system bus. If using a
+`yggdrasil` package, starting the `yggdrasil.service` unit should be sufficient.
+Be sure to configure `/etc/yggdrasil/config.toml` to connect to an MQTT broker,
+should you wish to send messages over the network.
 
-*HTTP server*: An optional HTTP server used to request payloads from localhost. This does not need to be more complicated than using the Python's `SimpleHTTPServer` module.
+## `mqttcli`
 
-# Install rhc-worker-playbook develop setup
+It is recommended to install a simple MQTT publish/subscribe utility. Included
+in Fedora is `mqttcli`, a pair of simple client utilities for such purpose.
 
-`rhc-worker-playbook` needs some devel packages in order to compile the C extensions, which may take some time to run as it dowloads and install everything needed to the `rhc-worker-playbook`.
+## HTTP server
 
-```console
-$ sudo dnf install c-ares-devel openssl-devel python3-devel gcc gcc-c++
-$ python -m venv .venv
-$ source .venv/bin/activate
-$ make build
-```
+An HTTP server is used to request payloads from localhost. This does not need to
+be more complicated than using the Python's `SimpleHTTPServer` module.
+
+# Compilation
 
-Additionally it can be used through `rhc` installing the `rhc-worker-playbook` as a worker.
+It is possible, though awkward, to run `rhc-worker-playbook` directly from the
+project repository. If you're working on a Fedora-derived distribution, use the
+included `srpm` meson target to create a SRPM. This SRPM can then be built using
+`mock` or `koji` or any other RPM build system.
 
 ```console
-$ sudo make CONFIG_DIR=$(pkg-config rhc --variable workerconfdir) installed-lib-dir
-$ sudo make build
-$ sudo make install CONFIG_DIR=$(pkg-config rhc --variable workerconfdir)
+meson setup -build_srpm=True builddir
```

Comment:
For some reason, this isn't working on my system:
```
usage: meson [-h] {setup,configure,dist,install,introspect,init,test,wrap,subprojects,rewrite,compile,devenv,env2mfile,help} ...
meson: error: unrecognized arguments: -build_srpm=True

```

Reply:
Whoops. This is a syntax error in my instructions.